testing/testnode_test.go

```diff
@@ -0,0 +1,15 @@
+package testing
+
+import "testing"
+
+// TestNewTestCluster is a smoke test for creating a cluster of Nodes for
+// testing and shutting them down.
+func TestNewTestCluster(t *testing.T) {
+	cluster, err := newTestCluster(t.TempDir())
```

Comment:
maybe the smoke test should assert the nodes can communicate with all components it needs to be able to communicate to? I'm asking because I think you mistakenly defer a `listener.Close()` in a constructor somewhere which means this cluster is effectively not working and still passing the smoke test.

Reply:
Yeah that's the next step once everyone agrees that this structure is fine.